tests/test_content_streams.py

```diff
@@ -204,3 +204,50 @@ async def test_empty_request():
 def test_invalid_argument():
     with pytest.raises(TypeError):
         encode(123)
+
+
+@pytest.mark.asyncio
+async def test_multipart_multiple_files_single_input_content():
+    files = [
+        ("file", io.BytesIO(b"<file content 1>")),
+        ("file", io.BytesIO(b"<file content 2>")),
+    ]
+    stream = encode(files=files, boundary=b"+++")
+    sync_content = b"".join([part for part in stream])
+    async_content = b"".join([part async for part in stream])
```

Comment:
Is there value in testing for the sync and async content concatenation here? I think we’re using async for in the other test cases.

(If there’s a missing test case for “async for yields the same than for” then we could add it, but separately to this PR imo.)

Reply:
I basically took the previous PR test and didn't think about it but looking all tests inside the module all have the very same joins on both sync and async.

Reply:
Ah, right, sorry then I didn’t actually check the code before writing that up. 😅 Either way works then, and we can clean up the tests later.